src/models/room-state.ts

```diff
@@ -30,6 +30,15 @@ import { Beacon, BeaconEvent, BeaconEventHandlerMap, getBeaconInfoIdentifier, Be
 import { TypedReEmitter } from "../ReEmitter";
 import { M_BEACON, M_BEACON_INFO } from "../@types/beacon";
 
+export interface IMarkerFoundOptions {
+    /** Whether the timeline was empty before the marker arrived in
```

Comment:
I take it back, there's 5 copies of this comment, and this is the first place that `the marker` makes sense. Can we please somehow DRY this to prevent 5 comment copies from falling out of sync.

Reply:
Remember an interface can extend many interfaces, so you can assemble the bigger ones really nicely

Reply:
I first tried drying things out by a [naive extending of everything](https://github.com/matrix-org/matrix-js-sdk/pull/2299/commits/58fa63c27a21aa2721b2c178caa7b2ea68cfca77) but the nature of this kinda sucks to follow (OOP hell).

I [updated it to use `extends Pick<Type, Keys>`](https://github.com/matrix-org/matrix-js-sdk/pull/2299/commits/d0466a1605d4ed68d49db5c4157c608a97137db0) instead. This way, we can better compose stuff together and share the comment from a single place and it's obvious which properties are defined for each interface (no need to follow the chain across files). This also avoids the problem of stuff unintentionally cascading across interfaces which have disparate concerns when we add properties in the future.

Seem reasonable?

Reply:
Yup, thanks!